proxy/http/HttpSM.cc

```diff
@@ -1954,9 +1954,7 @@ HttpSM::state_read_server_response_header(int event, void *data)
     t_state.api_next_action       = HttpTransact::SM_ACTION_API_READ_RESPONSE_HDR;
 
     // if exceeded limit deallocate postdata buffers and disable redirection
-    if (enable_redirection && (redirection_tries < t_state.txn_conf->number_of_redirections)) {
-      ++redirection_tries;
-    } else {
+    if (enable_redirection && (redirection_tries >= t_state.txn_conf->number_of_redirections)) {
```

Comment:
I don't think so this is a break of the multiple redirection, and make the post redirection cleanup, we are in progress of a redirection circle, and the SM is reseted, we should not check the old information here. and if we need to stop, we don't need to check the response anyway.